src/char/int_mercenary.cpp

```diff
@@ -90,6 +108,25 @@ bool mapif_mercenary_save(struct s_mercenary* merc)
 		flag = false;
 	}
 
+	// Save skill cooldowns
+	SqlStmt *stmt = SqlStmt_Malloc(sql_handle);
+
+	if (SQL_ERROR == SqlStmt_Prepare(stmt, "REPLACE INTO `%s` (`mer_id`, `skill`, `tick`) VALUES (%d, ?, ?)", schema_config.skillcooldown_mercenary_db, merc->mercenary_id))
+		SqlStmt_ShowDebug(stmt);
+	for (uint16 i = 0; i < MAX_SKILLCOOLDOWN; ++i) {
+		if (merc->scd[i].skill_id > 0 && merc->scd[i].tick != 0) {
+			SqlStmt_BindParam(stmt, 0, SQLDT_USHORT, &merc->scd[i].skill_id, 0);
+			SqlStmt_BindParam(stmt, 1, SQLDT_LONGLONG, &merc->scd[i].tick, 0);
+			if (SQL_ERROR == SqlStmt_Execute(stmt)) {
+				SqlStmt_ShowDebug(stmt);
+				SqlStmt_Free(stmt);
+				flag = false;
+				break;
+			}
+		}
```

Comment:
```suggestion
		if ( SQL_ERROR == SqlStmt_BindParam(stmt, 0, SQLDT_USHORT, &merc->scd[i].skill_id, 0)
		|| SQL_ERROR == SqlStmt_BindParam(stmt, 1, SQLDT_LONGLONG, &merc->scd[i].tick, 0)
		|| SQL_ERROR == SqlStmt_Execute(stmt)) {
			SqlStmt_ShowDebug(stmt);
			flag = false;
			break;
		}
```
Issues fixed:
Missing SQL_ERROR checks for the parameter binding
Double free of prepared statement.